capsules/src/button.rs

```diff
@@ -110,19 +94,9 @@ impl<'a> Button<'a> {
         }
     }
 
-    fn get_button_state(&self, pin_num: u32) -> ButtonState {
-        let index = pin_num as usize;
-        let pin_value = self.pins[index].0.read();
-        match self.pins[index].1 {
-            GpioMode::LowWhenPressed => match pin_value {
-                false => ButtonState::Pressed,
-                true => ButtonState::NotPressed,
-            },
-            GpioMode::HighWhenPressed => match pin_value {
-                false => ButtonState::NotPressed,
-                true => ButtonState::Pressed,
-            },
-        }
+    fn get_button_state(&self, pin_num: u32) -> gpio::ButtonState {
```

Comment:
I like this change and could see it being applied to the led capsule very similarly with the ActivationMode/State idea.